crates/fj-kernel/src/algorithms/sweep.rs

```diff
@@ -124,11 +124,22 @@ pub fn sweep_shape(
                 let top_edge =
                     source_to_top.edges().get(&edge_source).unwrap().clone();
 
-                let surface =
+                let surface = if path.dot(&Vector::from([0., 0., 1.]))
+                    >= fj_math::Scalar::from_f64(0.)
```

Comment:
Note that `Scalar` has some short-hands for often-used values, in this case `Scalar::ZERO`.